### PR TITLE
Feature: adding user repository

### PR DESCRIPTION
**Pull request by eapg**
* Adding Roles Enum
* Adding UserType Enum
* Adding Application.properties file with postgresql config
* Adding method `saveUsersInDb` as a example
* Adding User repository

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#--------------------- DB Connection ------------------
```
Comment (reynoldmorel) [inline src/main/resources/application.properties:1]: Move this to your local properties file

File: src/main/java/com/ea/restaurant/SpringBootApp.java [hunk 2]
```diff
@@ -2,8 +2,10 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 
 @SpringBootApplication
+@EnableJpaRepositories
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/SpringBootApp.java:8]: Move this to be above.

File: pom.xml [hunk 3]
```diff
@@ -34,13 +34,17 @@
...
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-validation</artifactId>
-            <version>3.0.1</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>42.5.1</version>
```
Comment (reynoldmorel) [inline pom.xml:47]: Use properties variables

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/main/java/com/ea/restaurant/SpringBootApp.java (hunk 2 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/SpringBootApp.java:8]: Done.

File: src/main/resources/application.properties (hunk 1 above)
Comment (eapg) [reply src/main/resources/application.properties:1]: done.

File: src/main/java/com/ea/restaurant/repository/UserRepository.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.ea.restaurant.repository;
+
+import com.ea.restaurant.entities.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {}
```
[the remark this answers is not in the document]
Comment (eapg) [reply src/main/java/com/ea/restaurant/repository/UserRepository.java]: Done.

File: src/main/java/com/ea/restaurant/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.User;
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+  User findByIdWithStatus(Long id);
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/repository/UserRepository.java:11]: This or not return a user, use `Optional<User>` instead.

Should be `findByIdAndStatus(String id, Status status)`

File: src/main/java/com/ea/restaurant/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+  User findByUsernameWithStatus(String username);
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/repository/UserRepository.java:13]: Same as above

File: src/main/resources/application-example.properties [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#--------------------- DB Connection ------------------
+spring.datasource.url=
+spring.datasource.username=
+spring.datasource.password=
+
+#--------------------JPA-ORM Properties-----------------
```
Comment (reynoldmorel) [inline src/main/resources/application-example.properties:6]: This should be in your main properties file, is there any reason why you moved this piece here?

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -0,0 +1 @@
+spring.profiles.active=local
```
Comment (reynoldmorel) [inline src/main/resources/application.properties:1]: Don't put this here, send this in the command you use to run the application

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/main/java/com/ea/restaurant/repository/UserRepository.java (hunk 5 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/repository/UserRepository.java:11]: Done.

File: src/main/java/com/ea/restaurant/repository/UserRepository.java (hunk 6 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/repository/UserRepository.java:13]: Done.

File: src/main/resources/application-example.properties (hunk 7 above)
Comment (eapg) [reply src/main/resources/application-example.properties:6]: Done.

File: src/main/resources/application.properties (hunk 8 above)
Comment (eapg) [reply src/main/resources/application.properties:1]: Done.

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**